### ISSUE TITLE
PIP version of bCNC does not run with python 3.10.2 on windows 10

### ISSUE THREAD

**Issue by cshunt** [vlachoudis/bCNC#1659]
After installing bCNC with python -m pip install bCNC, trying to run bCNC.bat or bCNC.py fails. "py -m bCNC" from cmd prompt.

\AppData\Local\Programs\Python\Python310\lib\site-packages\bCNC\Utils.py
NameError: name 'sys' is not defined

It appears this issue has been resolved in the master branch but not in a release yet. "Import sys" is missing from several files. It looks to be related to an update in python. utils.py, toolspage.py & filepage.py
adding "import sys" in these files lets bCNC run.. not sure if there are other modules that would crash because of this

**Comment by Git-Bruno** [comment]
I ran 'pip install bCNC' a whille back without problem with python 3.8 but I also get the error NameError: name 'sys' is not defined when I start bCNC with python 3.10 after installing bCNC with 'pip install bCNC'.
With 3.8 there is a directory  bCNC-0.9.14.317-py3.8.egg-info in Python38\Lib\site-packages\ after the install and there is an 'import sys' in modules utils.py, toolspage.py & filepage.py.  

With 3.10 there are errors compiling numpy (cl.exe failed with exit status 2 & ERROR: Failed building wheel for numpy) and when the pip command exists there is a bCNC-0.9.14.307-py3.10.egg-info in Python310\Lib\site-packages\. 
While the pip command runs I see what I interpret as an attempt to install different releases of bCNC down to bCNC-0.9.14.307 in which the 'import sys' statements are missing in modules utils.py, toolspage.py & filepage.py.  

After adding 'import sys' in modules utils.py, toolspage.py & filepage.py helps but then this is to run an old release of bCNC. No update done.
Below the last error messages I get. 
Collecting bCNC
  Using cached bCNC-0.9.14.316.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.315.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.314.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.313.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.312.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.311.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.310.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.309.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
  Using cached bCNC-0.9.14.308.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: opencv-python>=2.4 in c:\python310\lib\site-packages (from bCNC) (4.5.5.62)
Requirement already satisfied: pyserial<=3.0.1 in c:\python310\lib\site-packages (from bCNC) (3.0.1)

I am surprised to see a compile of numpy running as the version of numpy I have with 3.10 is higher than the minimum required :
>pip freeze
bCNC==0.9.14.307
numpy==1.22.1
opencv-python==4.5.5.62
packaging==21.3
Pillow==9.0.0
pip-review==1.1.1
pyparsing==3.0.6
pyserial==3.0.1
tk==0.1.0

The pip process is quite mysterious to me and I understand the attempt to recompile numpy is rather unexpected as the prereq for bCNC should be statisfied to begin with.  May be a packaging thing ?

**Comment by cshunt** [comment]
setup.py is requesting a specific version of opencv "opencv_version = '4.4.0.46'"
pip installs version opencv-python==4.5.5.62... If I change this in setup.py it appears to install properly. I would think that pip will succeed 

It is strange to me that pip decides to install cache versions of bCNC....

**Comment by Harvie** [comment]
I've uploaded fix by @cshunt to pypi as bCNC-0.9.14.318 please go ahead and test if this fixes issue for you.

**Comment by cshunt** [comment]
Hi @Harvie,
I tested on linux and window venv and both installed without issue, PIP and local install ran and pulled dependencies. I see a potential issue when open-cv gets updated. To keep similar checking for python2 compatibility, the version could be checked similar to below. I'm not sure if always using the latest version of open-cv could cause other issues.  

I will close the issue with this comment for now. Thanks for the update.

```
  if sys.version_info[0] >= 3:
	  #opencv_version = '4.5'#.5.62' # Recent version for Puthon 3
	  opencv_version = 'opencv-python>=4.5'
  else: #python version lower then 3 compatability
	  #opencv_version ='4.2.0.32' # use the last opencv version for python 2.7
	  opencv-python=='4.2.0.32' # use the last opencv version for python 2.7
```


```

		#'opencv-python==%s ; ("arm" not in platform_machine) and ("aarch64" not in platform_machine)' % (opencv_version),	#Note there are no PyPI OpenCV packages for ARM (Raspberry PI, Orange PI, etc...)
                opencv_version,
	],
```
